packages/runtime/agent-scheduler/src/leadershipManager.ts

```diff
@@ -0,0 +1,40 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { EventEmitter } from "events";
+import { IAgentScheduler } from "./agent";
+
+const leadershipTaskId = "leader";
+
+export class LeadershipManager extends EventEmitter {
+    private subscribed: boolean = false;
+
+    public get leader() {
+        return this.agentScheduler.pickedTasks().includes(leadershipTaskId);
+    }
+
+    constructor(private readonly agentScheduler: IAgentScheduler) {
+        super();
+        agentScheduler.on("picked", (taskId: string) => {
+            if (taskId === leadershipTaskId) {
```

Comment:
leadershipTaskId: I'd make it non-optional argument.
I.e. start forcing people to use unique key, even for legacy scenarios.
They may use "leader" today for all existing scenarios (for better back compat), but at least it forces new scenarios in Bohemia to start using other keys...

Reply:
I'll take a look at options here.  The advantage of this approach is that the API surface is pretty much 1:1 with what we currently offer (`.leader` property, `"leader"`/`"notleader"` events) making the migration easier.  Whereas with unique key that surface will change, requiring more effort to migrate.

Reply:
Latest update removes `LeadershipManager` and replaces it with `TaskSubscription` (which requires the user to specifiy the `taskId` but otherwise functions similarly).